sos/report/plugins/kafka.py

```diff
@@ -0,0 +1,69 @@
+# Copyright (C) 2024 Alejandro Santoyo <alejandro.santoyo@canonical.com>
+#
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+import os
+from sos.report.plugins import Plugin, UbuntuPlugin
+
+
+class Kafka(Plugin, UbuntuPlugin):
+    """
+    No current deb for Kafka but a tarball to download, decompress and
+    use. Two available snaps, i.e., `kafka` and `charmed-kafka`, with
+    the former not actively maintained anymore, so focusing on the
+    latter.
+    """
+
+    short_desc = 'Apache Kafka plugin'
+    plugin_name = 'kafka'
+    profiles = ('services',)
+    services = ('',)
+    is_snap = False
+
+    def _is_snap_installed(self):
+        if self.path_isdir('/var/snap/charmed-kafka'):
+            return True
+        return False
+
+    def setup(self):
+        self.is_snap = self._is_snap_installed()
```

Comment:
`snap` is a supported package manager in sos, so you can use the existing mechanisms for it. See the `maas` plugin for an example.

Reply:
ack, done, thanks